packages/docregistry/src/default.ts

```diff
@@ -154,11 +158,8 @@ export class DocumentModel
     this.dirty = true;
   }
 
-  private _onStateChanged(
-    sender: models.ISharedFile,
-    changes: models.NotebookChange | models.FileChange
-  ): void {
-    if (changes.contextChange || (changes as models.FileChange).sourceChange) {
+  private _onStateChanged(sender: ISharedFile, changes: DocumentChange): void {
+    if (changes.contextChange || (changes as FileChange).sourceChange) {
```

Comment:
I fully agree. I was surprised by that too. And indeed I removed it in #13317 because it has weird side effect. If a refresh is required when the path changes, the code should listen for `Context.pathChanged`.